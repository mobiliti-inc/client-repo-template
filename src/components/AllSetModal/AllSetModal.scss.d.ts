// This file is automatically generated.
// Please do not change this file!
interface CssExports {
  'browse-button': string;
  'content-description': string;
  'content-header': string;
  'fades-in-out': string;
  'modal': string;
  'modal-box': string;
  'modal-container': string;
  'modal-container--bordered': string;
  'modal-hide': string;
  'modal-icon-close': string;
  'modal-main-content': string;
  'modal-visible': string;
}
declare var cssExports: CssExports;
export = cssExports;
