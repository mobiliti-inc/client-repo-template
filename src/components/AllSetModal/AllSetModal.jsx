import React from 'react';
import { Link } from 'react-router-dom';
import classNames from 'classnames';

import { iconCloseWhite } from '../../assets/icons';

import styles from './AllSetModal.scss';

const AllSetModal = (props) => {
	const {
		modalIsVisible,
		headerText,
		bodyText,
		buttonText,
		customButton,
		buttonTo,
		customStyles,
		showCloseModal,
		bordered,
		onClose,
		showButton
	} = props;
	const modalVisibilityClass = modalIsVisible ? 'visible' : 'hide';

	return (
		<div styleName={`modal modal-${modalVisibilityClass}`}>
			<div styleName="modal-box">
				<div
					styleName={classNames('modal-container', { 'modal-container--bordered': bordered })}
					className={customStyles}
				>
					{showCloseModal && <img src={iconCloseWhite} alt="Close" role="presentation" styleName="modal-icon-close" onClick={onClose} />}
					<div styleName="modal-main-content">
						<h2 styleName="content-header">
							{headerText || 'One Last Step!'}
						</h2>

						<p styleName="content-description">
							{bodyText || ''}
						</p>
						{
							showButton && (customButton || (
								<Link styleName="browse-button" to={buttonTo || '/vehicles'}>
									{buttonText || 'Browse Vehicles'}
								</Link>
							))
						}
					</div>
				</div>
			</div>
		</div >);
};


AllSetModal.propTypes = {
	modalIsVisible: PropTypes.bool,
	headerText: PropTypes.string,
	bodyText: PropTypes.string,
	buttonText: PropTypes.string,
	customButton: PropTypes.node,
	buttonTo: PropTypes.string,
	showCloseModal: PropTypes.bool,
	bordered: PropTypes.bool,
	customStyles: PropTypes.string,
	onClose: PropTypes.func,
	showButton: PropTypes.bool
};

AllSetModal.defaultProps = {
	modalIsVisible: false,
	headerText: '',
	bodyText: '',
	buttonText: '',
	customButton: null,
	buttonTo: '',
	showCloseModal: false,
	bordered: false,
	customStyles: '',
	onClose: () => {},
	showButton: true
};

export default CSSModules(AllSetModal, styles, { allowMultiple: true });
