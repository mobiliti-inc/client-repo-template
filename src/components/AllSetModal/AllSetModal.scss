@import "../../styles/_variables.scss";

.modal {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  z-index: 10;
  background: $blue-overlay;

  &-box {
    height: 100%;
    width: 100%;
    background-color: transparent;
    display: flex;
    justify-content: center;
    align-items: center;
  }

  &-container {
    display: flex;
    color: black;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    background-color: $green-check;
    // background-image: url('../../assets//images/img_Green-Pattern.png');
    background-repeat: no-repeat;
    background-position: right top;
    height: 431px;
    width: 522px;
    position: relative;

    @media (max-width: $mobile-width) {
      height: 100%;
      width: 100%;
      padding: 10px;
    }

    &--bordered {
      border-radius: 5px;
      box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);
    }
  }

  &-visible {
    display: block;
  }

  &-hide {
    display: none
  }

  &-icon-close {
    position: absolute;
    top: 20px;
    right: 30px;
    cursor: pointer;
  }
}

.modal-main-content {
  display: flex;
  flex-direction: column;
  overflow-y: scroll;
  color: $white;
}

.content-header {
  color: $white;
  max-height: 31px;
  max-width: 303px;
  font-family: $futura;
  font-size: 34px;
  font-weight: bold;
  letter-spacing: -0.76px;
  line-height: 31px;
  margin-bottom: 2%;
  margin-top: 6%;
}

.content-description {
  max-height: 76px;
  max-width: 334px;
  font-family: $hanken-sans;
  font-size: 16px;
  line-height: 19px;
  margin: 0;
  margin-bottom: 35%;
}

.browse-button {
  display: flex;
  justify-content: center;
  align-items: center;
  font-family: $futura;
  font-size: 14px;
  font-weight: bold;
  line-height: 23px;
  cursor: pointer;
  height: 44px;
  width: 203px;
  text-decoration: none;
  border: none;
  box-shadow: 0 10px 20px 0 rgba(0, 0, 0, 0.12);
  background-color: $white;
  color: $green-check;
}