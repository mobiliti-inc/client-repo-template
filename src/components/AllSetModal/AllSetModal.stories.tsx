import { storiesOf } from '@storybook/react';
import { text, boolean } from '@storybook/addon-knobs';
import * as React from 'react';
import { AllSetModal } from '../';

storiesOf('AllSetModal', module)
	.add('default', () =>
		<AllSetModal
			headerText={text('headerText', 'Example header text')}
			bodyText={text('bodyText', 'Body Text')}
			buttonText={text('buttonText', 'Continue')}
			modalIsVisible={boolean('modalIsVisible', true)}
			bordered={boolean('bordered', false)}
			showCloseModal={boolean("showCloseModal", false)}
			showButton={boolean('showButton', false)}
		/>, {
			notes: 'Overlay can be gray, white or dark-gray'
		}
	);
