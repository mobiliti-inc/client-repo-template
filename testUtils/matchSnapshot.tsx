const { shallowToJson } = require('enzyme-to-json');

export const matchSnapshot = (subject: any) => {
	expect(shallowToJson(subject)).toMatchSnapshot();
};
