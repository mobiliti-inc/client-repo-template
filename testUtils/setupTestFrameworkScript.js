const React = require("react");
// const CSSModules = require("react-css-modules");
const Enzyme = require("enzyme");
const Adapter = require("enzyme-adapter-react-16");

const { shallow, render, mount } = Enzyme;

Enzyme.configure({ adapter: new Adapter() });
jest.setTimeout(30000);

global.React = React;
// global.CSSModules = CSSModules;
global.shallow = shallow;
global.render = render;
global.mount = mount;
