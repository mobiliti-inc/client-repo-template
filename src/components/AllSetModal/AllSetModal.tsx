import * as React from 'react';
import { Link } from 'react-router-dom';
import cx from 'classnames';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faTimes } from "@fortawesome/pro-light-svg-icons";

import './AllSetModal.scss';

interface AllSetModalProps {
	modalIsVisible?: boolean;
	headerText?: string;
	bodyText?: string;
	buttonText?: string;
	customButton?: React.ReactNode;
	buttonTo?: string;
	showCloseModal?: boolean;
	bordered?: boolean;
	customStyles?: string;
	onClose?: (...args: any[]) => any;
	showButton?: boolean;
}

const AllSetModal: React.FC<AllSetModalProps> = (props) => {
	const {
		modalIsVisible,
		headerText,
		bodyText,
		buttonText,
		customButton,
		buttonTo,
		customStyles,
		showCloseModal,
		bordered,
		onClose,
		showButton
	} = props;

	const modalVisibilityClass = modalIsVisible ? 'visible' : 'hide';

	return (
		<div styleName={`modal modal-${modalVisibilityClass}`}>
			<div styleName="modal-box">
				<div styleName={cx('modal-container', { 'modal-container--bordered': bordered })} className={customStyles}>
					{showCloseModal && <div styleName="modal-icon-close" onClick={onClose}><FontAwesomeIcon icon={faTimes} /></div>}
					<div styleName="modal-main-content">
						<h2 styleName="content-header">{headerText || 'One Last Step!'}</h2>

						{bodyText && <p styleName="content-description">{bodyText}</p>}
						{showButton &&
							(customButton || (
								<Link styleName="browse-button" to={buttonTo || '/vehicles'}>
									{buttonText || 'Browse Vehicles'}
								</Link>
							))}
					</div>
				</div>
			</div>
		</div>
	);
};

export default AllSetModal;