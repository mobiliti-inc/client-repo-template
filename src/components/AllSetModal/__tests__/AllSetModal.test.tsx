import React from 'react';
import { matchSnapshot } from '../../../../testUtils/matchSnapshot';
import { shallow } from 'enzyme';

import AllSetModal from '../AllSetModal';

describe('AllSetModal', () => {
	it('renders', () => {
		const tree = shallow(
			<AllSetModal />);
		matchSnapshot(tree);
	});

	it('should render custom header', () => {
		const tree = shallow(<AllSetModal headerText="Lol" />);
		expect(tree.find('h2').text()).toBe('Lol');
	});

	it('should render close button just in case needed', () => {
		const tree = shallow(<AllSetModal />);
		expect(tree.find('FontAwesomeIcon')).toHaveLength(0);
		tree.setProps({ showCloseModal: true });
		expect(tree.find('FontAwesomeIcon')).toHaveLength(1);
	});

	it('should be able to be hidden and show(toggled)', () => {
		const tree = shallow(<AllSetModal />);
		expect(tree.find('.modal-hide')).toHaveLength(1);
		expect(tree.find('.modal-visible')).toHaveLength(0);
		tree.setProps({ modalIsVisible: true });
		expect(tree.find('.modal-hide')).toHaveLength(0);
		expect(tree.find('.modal-visible')).toHaveLength(1);
	});

	it('should be able to be bordered', () => {
		const tree = shallow(<AllSetModal />);
		expect(tree.find('.modal-container--bordered')).toHaveLength(0);
		tree.setProps({ bordered: true });
		expect(tree.find('.modal-container--bordered')).toHaveLength(1);
	});
});
