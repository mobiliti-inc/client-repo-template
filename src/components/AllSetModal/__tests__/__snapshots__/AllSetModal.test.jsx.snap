// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AllSetModal renders 1`] = `
<div
  className="modal modal-hide"
>
  <div
    className="modal-box"
  >
    <div
      className="modal-container"
    >
      <div
        className="modal-main-content"
      >
        <h2
          className="content-header"
        >
          One Last Step!
        </h2>
        <p
          className="content-description"
        />
        <Link
          className="browse-button"
          replace={false}
          to="/vehicles"
        >
          Browse Vehicles
        </Link>
      </div>
    </div>
  </div>
</div>
`;
